src/Commands/ListPackages.php

```diff
@@ -42,7 +44,59 @@ public function handle()
             }
         }
 
+        if ($this->option('git')) {
+            $this->renderGitPackages($packages, $packages_path);
+
+            return;
+        }
+
         $headers = ['Package', 'Path'];
         $this->table($headers, $packages);
     }
+
+    protected function renderGitPackages($packages, $packages_path)
+    {
+        $gitPackages = collect($packages)
+            ->map(function ($package) use ($packages_path) {
+                return [
+                    'vendor' => $package[0],
+                    'name'   => $package[1],
+                    'path'   => $packages_path.implode(DIRECTORY_SEPARATOR, [$package[0], $package[1]]),
+                ];
+            })
+            ->filter(function ($package) {
+                return file_exists($package['path'].DIRECTORY_SEPARATOR.'.git');
+            })
+            ->map(function ($package) {
+                // Always run fetch first
+                (new Process('git fetch', $package['path']))->disableOutput()->run();
+
+                // get the amount of commits difference
+                $commitDifference = 0;
+                (new Process('git rev-list HEAD..origin --count', $package['path']))
+                    ->run(function ($type, $buffer) use (&$commitDifference) {
+                        $commitDifference = str_replace(["\n", "\r"], '', $buffer);
+                    });
+
+                // Get the current branch
+                $branch = null;
+                (new Process('git branch', $package['path']))
+                    ->run(function ($type, $buffer) use (&$branch) {
+                        if (Str::startsWith($buffer, '*')) {
+                            $branch = str_replace(["\n", "\r", ' ', '*'], '', $buffer);
+                        }
+                    });
+
+                return [
+                    $package['vendor'],
+                    $package['name'],
+                    $commitDifference,
+                    $branch,
+                ];
```

Comment:
This is pretty long and cryptic at first glance, so perhaps you could extract it to another function?

Reply:
I isolated the the commits difference & branch name retrieval into their own function with better comments. Hope that's enough :)